db/crud.go

```diff
@@ -854,41 +889,42 @@ func (db *Database) updateDoc(docid string, allowImport bool, expiry uint32, cal
 
 	// Update the document
 	if db.UseXattrs() {
-		err = db.Bucket.WriteUpdateWithXattr(key, KSyncXattrName, int(expiry), func(currentValue []byte, currentXattr []byte, cas uint64) (raw []byte, rawXattr []byte, deleteDoc bool, err error) {
+		var casOut uint64
+		casOut, err = db.Bucket.WriteUpdateWithXattr(key, KSyncXattrName, int(expiry), func(currentValue []byte, currentXattr []byte, cas uint64) (raw []byte, rawXattr []byte, deleteDoc bool, err error) {
 			// Be careful: this block can be invoked multiple times if there are races!
 			if doc, err = unmarshalDocumentWithXattr(docid, currentValue, currentXattr, cas); err != nil {
 				return
 			}
 
-			var updatedDoc *document
-			updatedDoc, _, shadowerEcho, err = documentUpdateFunc(doc, currentValue != nil)
+			docOut, _, shadowerEcho, err = documentUpdateFunc(doc, currentValue != nil)
```

Comment:
It's a bit confusing to have a `shadowerEcho` when presumably this code could execute in an XATTR-only scenario w/ no bucket shadowing.  I realize this was already in the master branch, but what do you think about renaming this to something that doesn't sound bucket-shadowing specific?